site/README.md

```diff
@@ -0,0 +1,54 @@
+# aip.dev static site generator
+
+Welcome to our NIH static site generator.
+
+**Note:** If you do not care about this and just want to write or modify AIPs,
+see `CONTRIBUTING.md`.
+
+## Why?
+
+GitHub Pages normally automatically builds documentation with [Jekyll][], but
+as the AIP system has grown, we are beginning to reach the limits of what
+Jekyll can handle:
+
+- AIP adoption is handled through fork-and-merge and top-down configuration
+  files will lead to repetitive merge conflicts.
+- Our grouping and listing logic has grown complicated, and had to be
+  maintained using complex and error-prone Liquid templates.
+- Jekyll is extensible but GitHub requires specific Jekyll plugins, meaning we
+  can not use off-the-shelf solutions for planned features (e.g. tabbed code
+  examples).
+- Lack of meaningful build CI caused failures.
+- Working with the development environment was (really) slow.
```

Comment:
Right, but this is primarily because we have broken out our AIPs into multiple files, which I do think is the correct interim solution. This significantly increases the number of "pages" that Jekyll needs to iterate through. I also suspect that it requires a lot more I/O to generate the site.

However, I still think the long term solution is a schema that supports a AIP spec (like OAS or RAML). The benefit of this approach is that the spec can be compiled into a single file (even if it doesn't start out that way). I feel that this could reduce the amount of time it takes to compile, and could be iteratively optimized (i.e. only update the spec where it's been changed, instead of recompiling from scratch each time.

Because more conventions are being enforced in the schema, I also think this might reduce our liquid templating (although the templating that remains may be more complicated).

Reply:
@rhamiltonsf and I discussed this on video and decided that the file breakup we planned is not quite the right solution, but that this gave us a clear path toward a better one (to be implemented shortly).